### ISSUE TITLE
Query param array insert only works with 4+ elements

### ISSUE THREAD

**Issue by RPuffer** [ibmdb/node-ibm_db#672]
So, I'm working on a project using `ibm_db` with an array input as a query param.  My SQL looks something like this:
```js
        const numArray = [ 1, 2, 3, 4 ]

	const results = await query(`
        SELECT DISTINCT
		cl.NUM
	FROM
		cl
		INNER JOIN cc
			ON cc.NUM = cl.NUM
	WHERE
		cl.OTHER_NUM = ?

	`,[
		numArray
	]);
```

I'm pulling in v2.7.0 as inserting arrays was addressed in that release.  But, when `numArray` contains less than 4 elements I get:
```
Error occurred while retrieving data: Wrong param format!
```
It seems to behave as expected when the `numArray` contains 4 or more elements, but anything less than 4 returns the above error. Not sure how that's possible 🤷 

I have a pretty gross looking workaround right now:
```js
	let input;
	switch (numArray.length) {
		case 0:
			return [];
		case 1:
			input = numArray[0];
			break;
		case 2:
			input = [ ...numArray, numArray[0], numArray[0] ];
			break;
		case 3:
			input = [ ...numArray, numArray[0] ];
			break;
		default:
			break;
	}

	const results = await query(`
        SELECT DISTINCT
		cl.NUM
	FROM
		cl
		INNER JOIN cc
			ON cc.NUM = cl.NUM
	WHERE
		cl.OTHER_NUM = ?

	`,[
		input
	]);
```
but hopefully this could be addressed in a future release

**Comment by bimalkjha** [comment]
@RPuffer It looks you are not using Array Insert at all. Are you sure that your script is able to insert all data in the table for 4+ elements?
Please check [test-array-insert.js](https://github.com/ibmdb/node-ibm_db/blob/master/test/test-array-insert.js) file to see how array insert works. For array insert, two things are mandatory:
* 1. Array parameters must be passed as object with `ParamType:"ARRAY"`.
* 2. For query APIs, ArraySize must be passed as argument.
OR, For execute() APIs, statement attribute `ibmdb.SQL_ATTR_PARAMSET_SIZE` must be set with arraysize.

So, for your test program, numArray must be set as below:
`const numArray = {ParamType:"ARRAY", DataType:1, Data:[ 1, 2, 3 ]};`
Then in query() API, you need to pass `ArraySize:3`. It should work.
Thanks.

**Comment by RPuffer** [comment]
Oh, okay, so this is a completely different issue than what was addressed in `2.7.0`.  So my title may be misleading then - the real issue is being able to use an array in a query parameter.  For some odd reason, when I use 4+ elements it does not complain about the query param format being invalid.  But, it also does not appear to be inserting the query param as I would expect.  From some further testing, it appears to just pluck the first element of the array and insert that into the query param, but only when there are 4+ elements

Basically, I want the end result to be something like this:
```sql
        SELECT DISTINCT
		cl.NUM
	FROM
		cl
		INNER JOIN cc
			ON cc.NUM = cl.NUM
	WHERE
		cl.OTHER_NUM IN (1,2,3,4)
```
I would prefer to use a query parameter if possible to avoid SQL injection risk.  Granted, I could format the SQL statements and sanitize the input myself, but I was hoping there was some way to do this with query params.

**Comment by bimalkjha** [comment]
@RPuffer Ok. I checked the code shared by you. The parameters should be an array but you are passing it as Array of Array. So, there is problem. You need to modify below part of code:
```javascript
	WHERE
		cl.OTHER_NUM = ?

	`,[
		numArray
	]);
```
to
```javascript
	WHERE
		cl.OTHER_NUM = ?

	`,  numArray );
```
Thanks.

**Comment by bimalkjha** [comment]
Also, I did not under stood that when your program is not for `insert` but for `select`, then why are you passing 4 values for single parameter marker(?) sign? Actually your code should be like below:
```javascript
        const numArray = [ 4 ];

	const results = await query(`
        SELECT DISTINCT
		cl.NUM
	FROM
		cl
		INNER JOIN cc
			ON cc.NUM = cl.NUM
	WHERE
		cl.OTHER_NUM = ?

	`, numArray );
```
i.e. number of values in array should be equal to the number of ? in sql. Thanks.

**Comment by RPuffer** [comment]
Yeah, i do want to do a SELECT, probably my mistake with the wording of my title.  But, the main problem is that `numArray` is not a constant size - I get that for normal params you want the number of params to match the number of `?`.  So, I'm wondering if there is a way to use an `IN` operator and query param for a variable length array.

So, if `numArray = [1,2,3]` 
I want the WHERE clause to be:
`WHERE cl.OTHER_NUM IN (1,2,3)`

and if `numArray = [1]`
I want the WHERE clause to be:
`WHERE cl.OTHER_NUM IN (1)`


If `numArray` always had a constant length this would be fairly trivial, but in the actual project it is a function input and it could have any number of elements.  It looks like this is a common SQL issue, but I'm not seeing any native solutions in most SQL libraries, so I'll likely have to manage this myself.


my current workaround is as follows:
```js
        const numArray = [ 1, 2, 3, 4 ];

	const results = await query(`
        SELECT DISTINCT
		cl.NUM
	FROM
		cl
		INNER JOIN cc
			ON cc.NUM = cl.NUM
	WHERE
		cl.OTHER_NUM IN (${new Array(numArray.length).fill('?').join(',')})
	`, numArray);
```
we could most likely close this issue as it doesn't seem very common to support this issue 🤷
